File: prime/src/main/java/com/telenordigital/prime/events/EventProcessorException.java
package com.telenordigital.prime.events;

import com.telenordigital.prime.storage.entities.NotATopupProductException;
import com.telenordigital.prime.storage.entities.PurchaseRequest;

public final class EventProcessorException extends Exception {

    private final PurchaseRequest pr;

    public EventProcessorException(final Throwable t) {
        super(t);
        this.pr = null;
    }

    public EventProcessorException(final String str, final PurchaseRequest pr) {
        super(str);
        this.pr = pr;
    }

    public EventProcessorException(
            final String str,
            final PurchaseRequest pr,
            final NotATopupProductException ex) {
        super(str, ex);
        this.pr = pr;
    }

    public EventProcessorException(final String str, final Throwable ex) {
        super(str, ex);
        this.pr = null;
    }

    @Override
    public String toString() {
        return super.toString() + ", pr = " + pr.toString();
    }
}

File: prime/src/main/kotlin/org/ostelco/prime/events/EventProcessorException.kt
package org.ostelco.prime.events

import org.ostelco.prime.storage.entities.NotATopupProductException
import org.ostelco.prime.storage.entities.PurchaseRequest
import java.util.*

class EventProcessorException : Exception {

    private val pr: PurchaseRequest?

    constructor(t: Throwable) : super(t) {
        this.pr = null
    }

    constructor(str: String, pr: PurchaseRequest) : super(str) {
        this.pr = pr
    }

    constructor(
            str: String,
            pr: PurchaseRequest,
            ex: NotATopupProductException) : super(str, ex) {
        this.pr = pr
    }

    constructor(str: String, ex: Throwable) : super(str, ex) {
        this.pr = null
    }

    override fun toString(): String {
        return super.toString() + ", pr = " + Objects.requireNonNull<PurchaseRequest>(pr).toString()
    }
}
